### PR TITLE
test: workflow templates integration tests

### PR DESCRIPTION
**Pull request by hatimkhan**
Issue #: GALI-719

This PR includes the tests for the workflow template drafts resources

Checklist:

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: main/integration-tests/__test__/api-tests/workflow-templates/create-workflow-template-version.test.js [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+      const templateId = setup.gen.string({ prefix: 'wt-template-test-create' });
+
+      await adminSession.resources.users.deactivateUser(researcherSession.user);
+
+      await expect(researcherSession.resources.workflowTemplates.versions(templateId).create()).rejects.toMatchObject({
+        code: errorCode.http.code.unauthorized,
+      });
+    });
+
+    it('should fail for internal guest', async () => {
```
Comment (Bingjiling) [inline main/integration-tests/__test__/api-tests/workflow-templates/create-workflow-template-version.test.js:53]: Do we want to use `it.each()` for internal guest, guest, and researcher?

File: main/integration-tests/__test__/api-tests/workflow-templates/get-workflow-templates.test.js [hunk 2]
```diff
@@ -0,0 +1,117 @@
...
+
+    it('should fail for researcher', async () => {
+      const researcherSession = await setup.createResearcherSession();
+      await expect(researcherSession.resources.workflowTemplates.get()).rejects.toMatchObject({
+        code: errorCode.http.code.forbidden,
+      });
+    });
+
+    it('should return workflow templates if admin', async () => {
+      await expect(adminSession.resources.workflowTemplates.get()).resolves.not.toHaveLength(0);
```
Comment (Bingjiling) [inline main/integration-tests/__test__/api-tests/workflow-templates/get-workflow-templates.test.js:71]: Can we create a workflow template here so the result is not empty?

File: main/integration-tests/support/complex/delete-workflow-template-version.js [hunk 3]
```diff
@@ -0,0 +1,107 @@
...
+
+const { run } = require('../utils/utils');
+
+/**
+ * A function that performs the complex task of deleting a workflow template version.
+ *
+ * @param {string} id The workflow template id to be deleted
+ * @param {number} v The version of the workflow template to be deleted
+ */
+async function deleteWorkflowTemplateVersion({ aws, id, version = 1 }) {
```
Comment (Bingjiling) [inline main/integration-tests/support/complex/delete-workflow-template-version.js:29]: This is a complicated function! Maybe it needs it's own test hahaha

**Review by Bingjiling — APPROVED** [review]
LGTM

File: main/integration-tests/__test__/api-tests/workflow-templates/get-workflow-templates.test.js (hunk 2 above)
Comment (hatimkhan) [reply main/integration-tests/__test__/api-tests/workflow-templates/get-workflow-templates.test.js:71]: There is a default one that must be there for the system to function. It the 'wt-empty'.  So we are checking that we have more than 0  (.not.toHaveLength(0))

File: main/integration-tests/__test__/api-tests/workflow-templates/create-workflow-template-version.test.js (hunk 1 above)
Comment (hatimkhan) [reply main/integration-tests/__test__/api-tests/workflow-templates/create-workflow-template-version.test.js:53]: I went back and forth on this. I felt it was easier to read this way.

**Pull request merged**